src/protocols/kubernetes/url.c

```diff
@@ -109,26 +110,52 @@ int guac_kubernetes_endpoint_attach(char* buffer, int length,
                 sizeof(escaped_pod), kubernetes_pod))
         return 1;
 
-    /* Generate attachment endpoint URL */
-    if (kubernetes_container != NULL) {
+    /* Generate endpoint path depending on the call type */
+    char* call = "attach";
+    if (exec_command != NULL)
+        call = "exec";
+
+    char endpoint_path[GUAC_KUBERNETES_MAX_ENDPOINT_LENGTH];
+
+    written = snprintf(endpoint_path, sizeof(endpoint_path), 
+        "/api/v1/namespaces/%s/pods/%s/%s", escaped_namespace, escaped_pod, call);
+        
+    if (written < 0 || written >= sizeof(endpoint_path))
+        return 1;
+
+    /* Generate endpoint params */
+    char endpoint_params[GUAC_KUBERNETES_MAX_ENDPOINT_LENGTH]="";
+
+    if (exec_command != NULL) {
+        /* Escape exec command */
+        if (guac_kubernetes_escape_url_component(escaped_exec_command,
+                    sizeof(escaped_exec_command), exec_command))
+            return 1;
 
+        written = snprintf(endpoint_params, sizeof(endpoint_params), 
+            "command=%s&", escaped_exec_command);
+
+        if (written < 0 || written >= sizeof(endpoint_params))
+            return 1;
+    }
+
+    if (kubernetes_container != NULL) {
         /* Escape container name */
         if (guac_kubernetes_escape_url_component(escaped_container,
                     sizeof(escaped_container), kubernetes_container))
             return 1;
 
-        written = snprintf(buffer, length,
-                "/api/v1/namespaces/%s/pods/%s/attach"
-                "?container=%s&stdin=true&stdout=true&tty=true",
-                escaped_namespace, escaped_pod, escaped_container);
-    }
-    else {
-        written = snprintf(buffer, length,
-                "/api/v1/namespaces/%s/pods/%s/attach"
-                "?stdin=true&stdout=true&tty=true",
-                escaped_namespace, escaped_pod);
+        written = snprintf(endpoint_params, sizeof(endpoint_params),
+                "container=%s&", escaped_container);
```

Comment:
Won't this overwrite the string previously written to `endpoint_params` if `exec_command` is set?

Reply:
Fixed